### DISCUSSION TITLE
Disable visualization to speed up simulation

### DISCUSSION THREAD

**Discussion by ValerioB88** [mesa/mesa#1122] [Q&A]
Hi, I am just getting started with mesa (great work btw!), so pardon me if my question is trivial. I am trying to understand if it's possible to disable the visualization (e.g with a shortcut/a button). With really dense simulations, we might just want to compute the model state with python, and only visualize it every once in a while, in order to speed up simulations when no graphics is needed. 

I have taken a look at the code and it's not obvious to me how to do that. I understand that the tornado part of the code wait for a response from the server before initiating the next step (or resetting the environment), so probably I need to act on that part somehow. Any suggestions?
Thanks

**Comment by rht** [comment]
See also #693, which has a description on how to run the model without the visualization.

**Comment by ValerioB88** [comment]
Thanks. I had in mind a different usecase. The simulation I have in mind would greatly benefit from visualization. The simulation is also very long, and it could last hours. I won't be in front of the computer the whole time, and during that time it's useless (and computationally expensive) have the server draw the graphics. I wanted to "interactively" disable visualization, maybe through a checkbox or shortcut. 
Is this something people would be interested in? Do you think it's useful? I would be happy to contribute back to the main project, since I'll probably end up coding this myself :)

**Reply by rht** [reply]
That is definitely useful. This is where the drawing happens: https://github.com/projectmesa/mesa/blob/68c48e6d1de2a3b7d102a3530920b2a4f5cc08e7/mesa/visualization/ModularVisualization.py#L208, where it can be disabled given a condition.
